python/cudf/cudf/utils/utils.py

```diff
@@ -212,3 +230,60 @@ def __array_function__(self, *args, **kwargs):
 
 
 IS_NEP18_ACTIVE = _is_nep18_active()
+
+
+def _set_rmm_config(
+    use_managed_memory=False,
+    use_pool_allocator=False,
+    initial_pool_size=None,
+    enable_logging=False,
+):
+    """
+    Parameters
+    ----------
+    use_managed_memory : bool, optional
+        If ``True``, use cudaMallocManaged as underlying allocator.
+        If ``False`` (default), use  cudaMalloc.
+    use_pool_allocator : bool
+        If ``True``, enable pool mode.
+        If ``False`` (default), disable pool mode.
+    initial_pool_size : int, optional
+        If ``use_pool_allocator=True``, sets initial pool size.
+        If ``None``, us
+es 1/2 of total GPU memory.
+    enable_logging : bool, optional
+        Enable logging (default ``False``).
+        Enabling this option will introduce performance overhead.
+    """
+    rmm.finalize()
+    rmm_cfg.use_managed_memory = use_managed_memory
+    if use_pool_allocator:
+        rmm_cfg.use_pool_allocator = use_pool_allocator
+        if initial_pool_size is None:
+            initial_pool_size = 0  # 0 means 1/2 GPU memory
+        elif initial_pool_size == 0:
+            initial_pool_size = 1  # Since "0" is semantic value, use 1 byte
+        if not isinstance(initial_pool_size, int):
+            raise TypeError("initial_pool_size must be an integer")
+        rmm_cfg.initial_pool_size = initial_pool_size
+    rmm_cfg.enable_logging = enable_logging
+    rmm.initialize()
+
+
+@initfunc
+def set_allocator(allocator="default", pool=False, initial_pool_size=None):
+    """
+    Set the GPU memory allocator. This function should be run only once,
+    before any cudf objects are created.
+    
```

Comment:
```suggestion

```

Reply:
@shwina if you leave the little "allow administrators to commit to this branch" box checked when you create your PRs, we can fix little details like this for you while you sleep, reducing PR latency during the endgame...

Reply:
I think we are able to push commits to the branch based on this line below.

```
Add more commits by pushing to the add-set-allocator branch on shwina/cudf.
```

That said, I don't think GitHub allows us to do this through the suggestion UI if we are not the PR author unfortunately. Have run into this before. 😞

FWIW also tried to add the suggested change, but it didn't let me do this either. 🤷‍♂

Reply:
Yea, github doesn't allow you to commit suggestions for some reason. You can still edit the file via the github UI or push commits directly to the branch without issue though.